images/sds-health-watcher-controller/src/pkg/controller/block_device_labels_watcher.go

```diff
@@ -0,0 +1,188 @@
+
+	for _, lvg := range lvgList.Items {
+		if len(lvg.Status.Nodes) == 0 {
+			log.Info(fmt.Sprintf("[ReconcileBlockDeviceLabels] LVMVolumeGroup %s nodes are not configured yet, retry later...", lvg.Name))
+			shouldRetry = true
+			continue
+		}
+
+		log.Debug(fmt.Sprintf("[ReconcileBlockDeviceLabels] tries to configure a selector from blockDeviceSelector of the LVMVolumeGroup %s", lvg.Name))
+		selector, err := metav1.LabelSelectorAsSelector(lvg.Spec.BlockDeviceSelector)
+		if err != nil {
+			return false, err
+		}
+		log.Debug(fmt.Sprintf("[ReconcileBlockDeviceLabels] successfully configured a selector from blockDeviceSelector of the LVMVolumeGroup %s", lvg.Name))
+
+		usedBdNames := make(map[string]struct{}, len(lvg.Status.Nodes[0].Devices))
+		for _, n := range lvg.Status.Nodes {
+			for _, d := range n.Devices {
+				usedBdNames[d.BlockDevice] = struct{}{}
+			}
+		}
+
+		shouldTrigger := false
+		if selector.Matches(labels.Set(blockDevice.Labels)) {
+			shouldTrigger = shouldTriggerLVGUpdateIfMatches(log, &lvg, blockDevice, usedBdNames)
+		} else {
+			shouldTrigger = shouldTriggerLVGUpdateIfNotMatches(log, &lvg, blockDevice, usedBdNames)
+		}
+
+		if shouldTrigger {
+			if lvg.Labels == nil {
+				lvg.Labels = make(map[string]string)
+			}
+			lvg.Labels[LVGUpdateTriggerLabel] = "true"
+			log.Info(fmt.Sprintf("[ReconcileBlockDeviceLabels] the LVMVolumeGroup %s should be triggered to update its configuration", lvg.Name))
+			err = cl.Update(ctx, &lvg)
+			if err != nil {
+				return false, err
+			}
+			log.Info(fmt.Sprintf("[ReconcileBlockDeviceLabels] successfully added the label %s to provide LVMVolumeGroup %s resource configuration update", LVGUpdateTriggerLabel, lvg.Name))
+		}
+	}
+
+	return shouldRetry, nil
+}
+
+func shouldTriggerLVGUpdateIfMatches(log logger.Logger, lvg *v1alpha1.LVMVolumeGroup, blockDevice *v1alpha1.BlockDevice, usedBdNames map[string]struct{}) bool {
+	log.Debug(fmt.Sprintf("[ReconcileBlockDeviceLabels] BlockDevice %s matches a blockDeviceSelector of the LVMVolumeGroup %s", blockDevice.Name, lvg.Name))
+	if _, used := usedBdNames[blockDevice.Name]; !used {
+		log.Info(fmt.Sprintf("[ReconcileBlockDeviceLabels] the BlockDevice %s matches the LVMVolumeGroup %s blockDeviceSelector, but is not used yet. Add the label %s to provide LVMVolumeGroup resource configuration update", blockDevice.Name, lvg.Name, LVGUpdateTriggerLabel))
+		return true
+	}
+
+	// for the case when BlockDevice stopped match the LVG blockDeviceSelector and then start again
+	for _, c := range lvg.Status.Conditions {
+		if c.Type == internal.TypeVGConfigurationApplied && c.Status == metav1.ConditionFalse {
+			log.Warning(fmt.Sprintf("[ReconcileBlockDeviceLabels] the BlockDevice %s matches the LVMVolumeGroup %s blockDeviceSelector, but the LVMVolumeGroup has condition %s in status False. Add the label %s to provide LVMVolumeGroup resource configuration update", blockDevice.Name, lvg.Name, c.Type, LVGUpdateTriggerLabel))
```

Comment:
> Add the label...

First I thought it's a recommendation to the reader of the logs. But then I realized it's a statement of what will be done on line 147.

There're already logs about the label on lines 148, 153, so may be let's remove "Add the label" part of the logs here and in two other places nearby